hathor/simulator/trigger.py

```diff
@@ -107,3 +110,32 @@ def __init__(self, sub_triggers: list[Trigger]) -> None:
 
     def should_stop(self) -> bool:
         return all(trigger.should_stop() for trigger in self._sub_triggers)
+
+
+class StopWhenSendLineMatch(Trigger):
+    """Stop the simulation when the node sends a line that matches a designated regex pattern.
+    """
+
+    def __init__(self, protocol: 'HathorLineReceiver', regex: 'Pattern') -> None:
+        # patches protocol.sendLine
+        self.original_send_line = protocol.sendLine
+        protocol.sendLine = self._send_line_wrapper  # type: ignore[method-assign]
```

Comment:
mypy fails here:

```
mypy -p hathor -p tests
hathor/simulator/trigger.py: note: In member "__init__" of class "StopWhenSendLineMatch":
hathor/simulator/trigger.py:122: error: Cannot assign to a method  [method-assign]
            protocol.sendLine = self._send_line_wrapper
            ^~~~~~~~~~~~~~~~~
```

Any idea on how to do it differently?

Reply:
Maybe this patching could be done through subclassing instead? Especially since we're moving away from simulator patches. Maybe you could use the existing builder scaffolding to do something like that, similar to how we did for the vertex verifiers. You would create a `SimulatorHathorLineReceiver` that would integrate with this trigger, instead of patching it manually.

If you deem that the current structure wouldn't support this, I'm ok with leaving the `ignore` comment.

Reply:
An alternative is using `setattr` (see https://github.com/python/mypy/issues/2427#issuecomment-480263443). I'm OK either way.

Reply:
I used `setattr()`. Thanks for your help!